Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1724,11 +1724,10 @@ void GridTrackSizingAlgorithm::run()
 
     // Step 3.
     m_strategy->maximizeTracks(tracks(m_direction), m_direction == GridTrackSizingDirection::ForColumns ? m_freeSpaceColumns : m_freeSpaceRows);
-    if (m_strategy->isComputingSizeOrInlineSizeContainment() && !m_renderGrid->explicitIntrinsicInnerLogicalSize(m_direction))
-        return;
 
     // Step 4.
-    stretchFlexibleTracks(initialFreeSpace);
+    if (!m_strategy->isComputingSizeOrInlineSizeContainment() || m_renderGrid->explicitIntrinsicInnerLogicalSize(m_direction))
```

Comment:
I would add a comment here describing in a little more detail what we are trying to do here as it is not super obvious, but overall this looks good.

Reply:
I do think the change is a little confusing, since De Morgan's law makes it seem like the new form is identical to the old.

Reply:
So I agree a better comment would be helpful, but otherwise this looks fine.